src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/TransmittedSingleton.java

```diff
@@ -0,0 +1,127 @@
+    }
+
+    @Override
+    public List<String> getKeyNames() {
+        return Arrays.asList(GATKVCFConstants.TRANSMITTED_SINGLETON, GATKVCFConstants.NON_TRANSMITTED_SINGLETON);
+    }
+    @Override
+    public Map<String, Object> annotate(final ReferenceContext ref,
+                                        final VariantContext vc,
+                                        final AlleleLikelihoods<GATKRead, Allele> likelihoods) {
+        Utils.nonNull(vc);
+        Set<Trio> trioSet = initializeAndGetTrios();
+        if (!vc.isBiallelic() || trioSet.isEmpty()) {
+            return Collections.emptyMap();
+        }
+        long highQualCalls = vc.getGenotypes().stream().filter(gt -> gt.getGQ() > HI_GQ_THRESHOLD).count();
+        if ((double) highQualCalls / vc.getNSamples() < CALL_RATE_THRESHOLD) {
+            return Collections.emptyMap();
+        }
+        final List<String> transmittedSingletonParent = new ArrayList<>();
+        final List<String> nonTransmittedSingletonParent = new ArrayList<>();
+        for (final Trio trio : trioSet) {
+            if (vc.isBiallelic() &&
+                    contextHasTrioGQs(vc, trio)) {
+
+                final boolean oneParentHasAllele = (vc.getGenotype(trio.getMaternalID()).isHet() && vc.getGenotype(trio.getPaternalID()).isHomRef()) || (vc.getGenotype(trio.getMaternalID()).isHomRef() && vc.getGenotype(trio.getPaternalID()).isHet());
+                final String matchingParentId = vc.getGenotype(trio.getMaternalID()).isHet() ? trio.getMaternalID() : trio.getPaternalID();
+
+                final boolean momIsHighGQ = vc.getGenotype(trio.getMaternalID()).getGQ() >= HI_GQ_THRESHOLD;
+                final boolean dadIsHighGQ = vc.getGenotype(trio.getPaternalID()).getGQ() >= HI_GQ_THRESHOLD;
+
+                final boolean childIsHighGQHet = vc.getGenotype(trio.getChildID()).isHet() && vc.getGenotype(trio.getChildID()).getGQ() >= HI_GQ_THRESHOLD;
+                final boolean childIsHighGQHomRef = vc.getGenotype(trio.getChildID()).isHomRef() && vc.getGenotype(trio.getChildID()).getGQ() >= HI_GQ_THRESHOLD;
+
+                final boolean childIsHighDepth = vc.getGenotype(trio.getChildID()).getDP() >= HI_DP_THRESHOLD;
+                final boolean momIsHighDepth = vc.getGenotype(trio.getChildID()).getDP() >= HI_DP_THRESHOLD;
+                final boolean dadIsHighDepth = vc.getGenotype(trio.getChildID()).getDP() >= HI_DP_THRESHOLD;
+
+                if (childIsHighDepth && momIsHighDepth && dadIsHighDepth &&
+                        vc.getAttributeAsInt(VCFConstants.ALLELE_COUNT_KEY, 0) == 2 &&
+                        childIsHighGQHet && oneParentHasAllele && momIsHighGQ && dadIsHighGQ) {
+                        transmittedSingletonParent.add(matchingParentId);
+                }
+                //TODO: This only works for trios (not quads or other more complicated family structures that would effect number of singletons for parents or transmission to multiple kids)
```

Comment:
what happens if the pedigree file has 2 kids in it? Will this clobber one/another? Should we add some defensive behavior to make sure this doesn't break anything?

Reply:
We have lots of singleton loggers putting warnings out about annotation warnings like this. IF you wanted to add a "detected a quad pedigree this will cause xyz problems" and we should probably at least have some tests codifying the behavior.

Reply:
I made this all clearer and added a warning and a test.